refreshable/refreshable_validating.go

```diff
@@ -41,6 +27,22 @@ func (v *ValidatingRefreshable) LastValidateErr() error {
 // validatingFn successfully. This returns an error if the current value of the passed in Refreshable does not pass the
 // validatingFn or if the validatingFn or Refreshable are nil.
 func NewValidatingRefreshable(origRefreshable Refreshable, validatingFn func(interface{}) error) (*ValidatingRefreshable, error) {
+	mappingFn := func(i interface{}) (interface{}, error) {
+		if err := validatingFn(i); err != nil {
+			return nil, err
+		}
+		return nil, nil
+	}
+	return newValidatingRefreshable(origRefreshable, mappingFn, false)
+}
+
+// MapValidatingRefreshable is similar to NewValidatingRefreshable but allows for the function to return a mapping/mutation
+// of the input object in addition to returning an error. It must always return a value of same type.
+func MapValidatingRefreshable(origRefreshable Refreshable, mappingFn func(interface{}) (interface{}, error)) (*ValidatingRefreshable, error) {
+	return newValidatingRefreshable(origRefreshable, mappingFn, true)
+}
+
+func newValidatingRefreshable(origRefreshable Refreshable, validatingFn func(interface{}) (interface{}, error), storeVal bool) (*ValidatingRefreshable, error) {
```

Comment:
nit: `s/storeVal/storeMappedVal` to be more explicit about what we're storing?

Reply:
Done.